extensions/amp-form/0.1/amp-form.js

```diff
@@ -251,6 +257,13 @@ export class AmpForm {
 
   /** @private */
   installEventHandlers_() {
+    this.viewer_.whenFirstVisible().then(() => {
```

Comment:
we probably want autofocus to take effect everytime document becomes visible (think viewer swipes). `whenNextVisible` does that (and covers first visibility as well)

Reply:
Ah you're right; I thought it was just once after the next time it became visible.

Reply:
Is the document state reset between viewer swipes? i.e. would the user expect the field to autofocus again if they swipe to the next document and then swipe back?

Reply:
oh naming! :) if you can think of a shorter version of `ifAlreadyVisibleOrWhenNextVisible` , we can rename. it is a fairly new feature and not used in many places yet.

Reply:
haha yeah one of the hardest problems ; )